test/common/upstream/hds_helper.h

```diff
@@ -0,0 +1,18 @@
+#include "common/upstream/health_discovery_service.h"
+
+#include "test/mocks/upstream/mocks.h"
```

Comment:
I think if you make `HdsInfoFactory` an interface under `include/envoy/upstream` (recommend renaming to `ClusterInfoFactory` for generality). then you can add the mock implementation for this in `tests/mocks/upstream`. That would be the more usual organization in Envoy for mocked implementations. You could actually avoid defining any implementation methods by using mock methods like in https://github.com/envoyproxy/envoy/blob/master/test/mocks/upstream/mocks.h#L146.